server/hotrod/src/main/java/org/infinispan/server/hotrod/tx/ServerTransactionOriginatorChecker.java

```diff
@@ -0,0 +1,50 @@
+package org.infinispan.server.hotrod.tx;
+
+import java.util.Collection;
+import java.util.Map;
+
+import org.infinispan.factories.annotations.Inject;
+import org.infinispan.remoting.rpc.RpcManager;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.transaction.impl.TransactionOriginatorChecker;
+import org.infinispan.transaction.xa.GlobalTransaction;
+
+/**
+ * A {@link TransactionOriginatorChecker} implementation that is aware of client transactions.
+ * <p>
+ * The transaction originator in this case is the Hot Rod client.
+ *
+ * @author Pedro Ruivo
+ * @since 9.1
+ */
+public class ServerTransactionOriginatorChecker implements TransactionOriginatorChecker {
+
+   private final Map<CacheXid, TxState> globalTxTable;
+   private RpcManager rpcManager;
+
+   public ServerTransactionOriginatorChecker(
+         Map<CacheXid, TxState> globalTxTable) {
+      this.globalTxTable = globalTxTable;
+   }
+
+   @Inject
+   public void inject(RpcManager rpcManager) {
+      this.rpcManager = rpcManager;
+   }
+
+   @Override
+   public boolean isOriginatorMissing(GlobalTransaction gtx) {
+      return isOriginatorMissing(gtx, rpcManager.getMembers());
+   }
+
+   @Override
+   public boolean isOriginatorMissing(GlobalTransaction gtx, Collection<Address> liveMembers) {
+      return !liveMembers.contains(gtx.getAddress()) && isNonClientTransaction(gtx);
+   }
+
+   private boolean isNonClientTransaction(GlobalTransaction gtx) {
+      return globalTxTable.values().stream()
```

Comment:
that will break the transactions. GlobalTransaction needs to be unique and your suggestion would make all nodes to create the GlobalTransaction with the same Address. 
I guess I can create a client address per server/cache manager if that is ok for you. I'm open to other suggestions as well :)